docs/src/math.md

```diff
@@ -0,0 +1,69 @@
+# On the Math of ChainRules.jl
+This page explain the mathematical underpinnings of ChainRulesCore.jl.
+Primarily in the abstract.
+It should not be read as documentation on how things work,
+but rather on why things work the way they do.
+
+It is written around the Julia idea of types.
+Which do not nessicarily correspond too closely with the idea of Programming Language Theory (static) types,
+nor with type-theory types.
+The key feature of the the Julia type system is that it enables multiple dispatch.
+
+As Julia is a dynamic programming language functions can return different types of values depending on the values of their inputs (i.e. not be type stable).
+
+Bacause of multiple dispatch,
+all functions are able to be define different _methods_
+dependencing on the types of the inputs.
+We thus can have defined different ``+``
+for different input types, and so will not distinguish between them.
+The functions thus stand alone from the types,
+except that a set of input type may special case them.
+Where as most similar definitions might decribe a object as a type and some operations on the type,
+we can consider them seperately.
+
+!!! terminology Notation:
```

Comment:
This block doesn't seem to render correctly in the PR-preview

Reply:
may need to be 
```
!!! terminology "Notation"
```

Reply:
```suggestion
!!! terminology "Notation"
```